webpack/scenes/Subscriptions/Details/SubscriptionDetails.js

```diff
@@ -18,12 +19,22 @@ class SubscriptionDetails extends Component {
     // eslint-disable-next-line react/prop-types
     const routerParams = this.props.match.params;
     this.props.loadSubscriptionDetails(parseInt(routerParams.id, 10));
+    this.props.loadProducts({
+      subscription_id: parseInt(routerParams.id, 10),
```

Comment:
I wonder if `parseInt(routerParams.id, 10)` can be moved to a constant somewhere since we keep repeating it in this file?

Reply:
That's a good thought, maybe a helper named `parseBaseTen()` or something?

Reply:
I more meant for the subscription id that its getting, have a const like `subscriptionId` that can be used.

Reply:
Ah, that makes more sense

Reply:
I'm willing to change this, but I also think that we might refactor this to come up with a solution that is reusable (as Walden was suggesting), especially since Foreman has to do these types of lookups as well.

Reply:
I would have to see what you mean - I don't see any value in abstracting out `parseInt` as that is a single function and `routerParams` is specific to this component  which is why I suggested refactoring just in this component

Reply:
`routerParams` is just `this.props.match.params`. We're just parsing the passed in ID (which is a string that we need to make an integer) to the GET request. This attribute is common to all our APIs (afaik), so I'd think Foreman is doing something similar to solve this same problem.

Reply:
I'm open to suggestions! I was just thinking something like this (or similar) when I started the discussion:

```javascript
getSubscriptionId() {
  const routerParams = this.props.match.params;
  return parseInt(routerParams.id, 10);
}
```